cmd/agent/subcommands/run/command.go

```diff
@@ -498,6 +501,14 @@ func startAgent(
 		} else {
 			// Subscribe to `AGENT_TASK` product
 			rcclient.SubscribeAgentTask()
+
+			if pkgconfig.Datadog.GetBool("remote_configuration.agent_integrations.enabled") {
+				// Spin up the config provider to schedule integrations through remote-config
+				rcProvider := providers.NewRemoteConfigProvider()
+				rcclient.Subscribe(data.ProductAgentIntegrations, rcProvider.IntegrationScheduleCallback)
+				// LoadAndRun is called later on
+				common.AC.AddConfigProvider(rcProvider, true, 10*time.Second)
```

Comment:
Would it make sense to handle uniformly with other configuration providers [here](https://github.com/datadog/datadog-agent/blob/2df07695cccd5273045bd6b08067cce2438dcadf/cmd/agent/common/autodiscovery.go#L117)?

This would allow RC checks to be executed with the `agent check` command, which is important for troubleshooting.

There are also common options to enable and disable configuration providers, so perhaps a separate `enabled` setting is not needed.

Reply:
> so it's a bit of a chicken and egg problem here

I don't think this is a problem, fx.Invoke can pass a component to the callback: 

```
		fx.Invoke(func(lc fx.Lifecycle, rcClient rcclient.Component) {
```

In other places where `LoadComponents` is called components are already initialized, so it should be a matter of simply adding a module and a parameter.